addons/survey/models/survey_user.py

```diff
@@ -267,240 +267,96 @@ def write(self, vals):
-    def save_line_simple_choice(self, user_input_id, question, post, answer_tag):
-        vals = {
-            'user_input_id': user_input_id,
-            'question_id': question.id,
-            'survey_id': question.survey_id.id,
-            'skipped': False
-        }
-        old_uil = self.search([
-            ('user_input_id', '=', user_input_id),
-            ('survey_id', '=', question.survey_id.id),
-            ('question_id', '=', question.id)
-        ])
-        old_uil.sudo().unlink()
-
-        if answer_tag in post and post[answer_tag].strip():
-            vals.update({'answer_type': 'suggestion', 'value_suggested': int(post[answer_tag])})
-        else:
-            vals.update({'answer_type': None, 'skipped': True})
-
-        # '-1' indicates 'comment count as an answer' so do not need to record it
-        if post.get(answer_tag) and post.get(answer_tag) != '-1':
-            self.create(vals)
-
-        comment_answer = post.pop(("%s_%s" % (answer_tag, 'comment')), '').strip()
-        if comment_answer:
-            vals.update({'answer_type': 'text', 'value_text': comment_answer, 'skipped': False, 'value_suggested': False})
-            self.create(vals)
+    def _save_line_matrix(self, vals, old_answers, answers, comment):
+        vals_list = []
 
-        return True
-
-    @api.model
-    def save_line_multiple_choice(self, user_input_id, question, post, answer_tag):
-        vals = {
-            'user_input_id': user_input_id,
-            'question_id': question.id,
-            'survey_id': question.survey_id.id,
-            'skipped': False
-        }
-        old_uil = self.search([
-            ('user_input_id', '=', user_input_id),
-            ('survey_id', '=', question.survey_id.id),
-            ('question_id', '=', question.id)
-        ])
-        old_uil.sudo().unlink()
-
-        ca_dict = dict_keys_startswith(post, answer_tag + '_')
-        comment_answer = ca_dict.pop(("%s_%s" % (answer_tag, 'comment')), '').strip()
-        if len(ca_dict) > 0:
-            for key in ca_dict:
-                # '-1' indicates 'comment count as an answer' so do not need to record it
-                if key != ('%s_%s' % (answer_tag, '-1')):
-                    val = ca_dict[key]
-                    vals.update({'answer_type': 'suggestion', 'value_suggested': bool(val) and int(val)})
-                    self.create(vals)
-        if comment_answer:
-            vals.update({'answer_type': 'text', 'value_text': comment_answer, 'value_suggested': False})
-            self.create(vals)
-        if not ca_dict and not comment_answer:
-            vals.update({'answer_type': None, 'skipped': True})
-            self.create(vals)
-        return True
+        for row_key, row_answer in answers.items():
+            for answer in row_answer:
+                vals.update({'answer_type': 'suggestion', 'value_suggested': answer, 'value_suggested_row': row_key})
+                vals_list.append(vals.copy())
```

Comment:
This is a little weird.

Can't you simply do:

`vals_list.append({'answer_type': 'suggestion', 'value_suggested': answer, 'value_suggested_row': row_key})` ?

Reply:
nope, because vals comes from caller
and is composed of : 
```
vals = {
            'user_input_id': user_input_id,
            'question_id': question.id,
            'survey_id': question.survey_id.id,
            'skipped': False,
        }
```